.github/workflows/makecode.yml

```diff
@@ -0,0 +1,94 @@
+name: Build MakeCode
+
+# MakeCode has to git clone/checkout this repo commit SHA, so only run this
+# workflow on pushes, as PRs generate a "merge commit" that is not clonable.
+on:
+  push:
+    branches: '*'
```

Comment:
Do we want this to run on _all_ branch pushes, or just `master` ?

Reply:
All branches, otherwise we'll merge something to master without knowing if it works in MakeCode.

Reply:
I might be wrong, but isn't the `branches` selection here the _destination_ branch, not the source? So we'd just need `master` ?

Reply:
Yes, so if the workflow is configure to only run on "master" and somebody is working in a feature branch this workflow will not run until it is merged into master (also remember that it doesn't run on PRs either). So a feature branch could be breaking MakeCode but we would only find out after it's been merged in the master branch, as that would be the first time this workflow would run.

Reply:
Ah, sorry I misread this as applying to PRs, Ok, happy to merge then.

Reply:
Yeah, the mean problem with PRs is that GitHub will create a merge commit to test the PR as it'd be when merged with the target branch. And this commit hash is not really pushed anywhere, so it cannot be "git cloned" or "git checkout" by build.py, MakeCode, nor MicroPython, and it's only accessible when using the `actions/checkout` in a step in the workflow.

So cannot run this workflow on PRs as MakeCode has no way to access that codal commit hash.